realsense_ros_camera/src/realsense_node_factory.cpp

```diff
@@ -29,47 +31,69 @@ RealSenseNodeFactory::RealSenseNodeFactory()
     rs2::log_to_console(severity);
 }
 
-void RealSenseNodeFactory::onInit()
+rs2::device RealSenseNodeFactory::getDevice(std::string& serial_no)
 {
-    try{
-        auto list = _ctx.query_devices();
-        if (0 == list.size())
+    auto list = _ctx.query_devices();
+    if (0 == list.size())
```

Comment:
nit: if its a vector could use `.empty()` instead of checking for 0

Reply:
This was how it was implemented upstream in the intel ros package. So I'm not gonna change it. But agreed this is a unconventional implementation.